### ISSUE TITLE
6 holes (aka variable hole tablets)

### ISSUE THREAD

**Issue by CableGuy67** [Shelagh-Lewins/twisted-threads#21]
I'm going to guess this one will have to be labeled, "for future"

Currently each tablet can have a maximum of 4 holes. Grante, this gives the ability to to 2 hole patterns (and I guess 3 hole) by using the amount of turns per pick. I've not tried it to see if it really works that way or if you are stuck with an empty hole sort of pattern. I'm guessing the latter.

Should you decide that this is a worthwhile it would make sense then to consider the differences with the variable sheds. Are you using the center shed and tablet with a flat upward? A point upward and one of the minor sheds? From my initial tests the difference is enough that the patterns are not identical. In fact, they can vary quite a bit with 6 hole tablets.

Tag this as you see fit. Mostly I just put it here so the idea wasn't lost entirely.

**Comment by Shelagh-Lewins** [comment]
I'm currently thinking to allow 2, 4 and 6 holes, limiting it because I've no idea how odd numbers of holes would function. I think 3 and 5 would come out as missed hole; I am not sure how common they are?

Good thought about variable sheds, though it would again require me to find out how the weaving works!

**Comment by CableGuy67** [comment]
Yeah, from what I can tell triangular tablets aren't really that common probably due to the difficulty of working with them. I'll be glad to do some test weaving with hex cards once you start getting things together enough to test though. Perhaps that will shed some light on the best way to display them in the visualizer. Mostly I fall back to B.Datta's work when imagining how things will look.
https://www.tabletweavers.org/B.Datta_monograph.php

It's nice to see that the rewrite is beginning. I will be happy to test things along the way once you are to that point.

**Comment by CableGuy67** [comment]
Ah, so yes, 2, 4 and 6 (possibly even 8) would most likely cover 99% of the use cases anyhow.
